libr/core/cmd_search.inc.c

```diff
@@ -288,25 +301,24 @@ static int search_hash(RCore *core, const char *hashname, const char *hashstr, u
 			}
 			R_LOG_INFO ("Search in range 0x%08"PFMT64x " and 0x%08"PFMT64x, from, to);
 			int blocks = (int) (to - from - len);
-			R_LOG_INFO ("Carving %d blocks:", blocks);
+			R_LOG_INFO ("Carving %d blocks", blocks);
 			(void) r_io_read_at (core->io, from, buf, bufsz);
 			for (i = 0; (from + i + len) < to; i++) {
 				if (r_cons_is_breaked ()) {
 					break;
 				}
 				char *s = r_hash_tostring (NULL, hashname, buf + i, len);
-				if (!(i % 5)) {
-					eprintf ("%d\r", i);
-				}
+				print_search_progress (i, to, 0, param);
 				if (!s) {
 					R_LOG_ERROR ("Hash fail");
 					break;
 				}
-				// eprintf ("0x%08"PFMT64x" %s\n", from+i, s);
 				if (!strcmp (s, hashstr)) {
-					R_LOG_INFO ("Found at 0x%"PFMT64x, from + i);
-					r_cons_printf ("f hash.%s.%s = 0x%"PFMT64x "\n",
-						hashname, hashstr, from + i);
+					char hash_short[9];
+					strncpy (hash_short, hashstr, 8);
+					r_strf_var (flag, 256, "%s.%s", hashname, hash_short);
+					r_cons_printf ("0x%" PFMT64x ": %s : %s\n", from + i, flag, hashstr);
+					r_flag_set (core->flags, flag, from + i, strlen (hashstr));
```

Comment:
maybe the size should be strlen/2?

Reply:
Right, in fact it is more appropriate to give the block length which was hashed I think.